File: src/main/java/thedarkcolour/futuremc/block/BlockCornflower.java
package thedarkcolour.futuremc.block;

import net.minecraft.block.SoundType;
import net.minecraft.init.Biomes;
import net.minecraft.world.biome.Biome;
import thedarkcolour.futuremc.init.FutureConfig;

public class BlockCornflower extends BlockFlower { //Adds blue flower
    public BlockCornflower() {
        super("FlowerBlue");
        setSoundType(SoundType.PLANT);
    }

    @Override
    public boolean isBiomeValid(Biome biome) {
        return biome == Biomes.PLAINS || biome == Biomes.MUTATED_FOREST;
    }

    @Override
    public int getFlowerChance() {
        return FutureConfig.modFlowers.cornflowerChance;
    }
}

File: src/main/java/thedarkcolour/futuremc/block/BlockCornflower.kt
package thedarkcolour.futuremc.block

import net.minecraft.init.Biomes
import net.minecraft.world.biome.Biome
import thedarkcolour.core.util.matchesAny
import thedarkcolour.futuremc.config.FConfig

class BlockCornflower : BlockFlower("cornflower") {
    override fun isBiomeValid(biome: Biome): Boolean {
        return biome.matchesAny(Biomes.PLAINS, Biomes.MUTATED_FOREST)
    }

    override val flowerChance: Double
        get() = FConfig.villageAndPillage.cornflower.spawnRate
}